Sources/Commands/SwiftTestTool.swift

```diff
@@ -133,6 +166,22 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
     }
 
     override func runImpl() throws {
+        
+        if let workers = options.numberOfWorkers {
+            
+            guard options.mode == .runParallel else {
+                // if user called --num-workers without the --parallel option, emit a diagnostic
+                diagnostics.emit(data: RequiredArgumentDiagnostic(requiredArgument: "--parallel", dependentArgument: "--num-workers"))
+                return
+            }
+            
+            // If --num-workers gets called with zero or less, emit a diagnostic
+            if workers <= 0 {
+                diagnostics.emit(data: UnsatisfiedIntParameterDiagnostic())
+                return
+            }
+        }
```

Comment:
We can factor this into a new method. Something like:

```swift
func validateArguments() throws {

    // Validation for --num-workers.
    if let workers = options.numberOfWorkers {

        // The --num-worker option should be called with --parallel.
        guard options.mode == .runParallel else {
            diagnostics.emit(
                data: RequiredArgumentDiagnostic(requiredArgument: "--parallel", dependentArgument: "--num-workers"))
            throw Diagnostics.fatalError
        }

        guard workers > 0 {
            diagnostics.emit(data: InvalidNumWorkersValueDiagnostic())
            throw Diagnostics.fatalError
        }
    }
}
```

Reply:
Perfect. 
Is `error: 
error: fatalError` preferred over just `error: ` ?

Reply:
Diagnostics.fatalError is a dummy error to abort the execution. DiagnosticEngine will take care of printing the actual error

Reply:
But I get 
![image](https://user-images.githubusercontent.com/18747072/43039988-ed664c50-8d06-11e8-9f41-8e712b9133a4.png)

is that correct?

Reply:
hmm, that sounds a bug. We shouldn't be printing the dummy error. Can you file a JIRA for that?

Reply:
Sure!